sw/device/lib/handler.h

```diff
@@ -16,4 +16,59 @@ typedef enum exc_id {
   kIdMax = 31
 } exc_id_t;
 
+/**
+ * Default exception handler. Can be overidden.
+ */
+__attribute__((aligned(4), interrupt)) void handler_exception(void);
```

Comment:
As more of an aside: can we force the `interrupt` attribute to force alignment for these?  We can never have an unaligned `interrupt`-attributed function.

Reply:
I shall look at adding this to clang. I'm not sure we can ask for it for gcc.

Reply:
Not entirely sure what making these handlers "weak" adds, as these are top level set of handlers that the CPU jumps to when receives an interrupt. Hence they should be implemented in a central place, which they are. If there are specific use-cases when these handlers might need to be overridden, could you name them please.

Reply:
Making them weak makes writing software without some kind of centralized IRQ handling easy.